File: csharp/ToolGood.Algorithm2/LitJson/JsonException.cs
using System;


namespace ToolGood.Algorithm.LitJson
{
    class JsonException : ApplicationException
    {

        internal JsonException(ParserToken token, Exception inner_exception) :
            base(String.Format("Invalid token '{0}' in input string", token), inner_exception)
        {
        }

        internal JsonException(int c) :
            base(String.Format("Invalid character '{0}' in input string", (char)c))
        {
        }


        public JsonException(string message) : base(message)
        {
        }

    }
}

File: java/toolgood.algorithm/src/main/java/toolgood/algorithm/litJson/JsonException.java
package toolgood.algorithm.litJson;

public class JsonException extends Exception {
    /**
     *
     */
    private static final long serialVersionUID = 1L;

    public JsonException(ParserToken token, Exception inner_exception) {
        super();
        // base(String.Format("Invalid token '{0}' in input string", token),
        // inner_exception)
    }

    public JsonException(int c) {
        super();
        // base(String.Format("Invalid character '{0}' in input string", (char)c))
    }

    public JsonException(String message) {
        // : base(message)
    }
}